Framework/Crystal/src/OptimizeLatticeForCellType.cpp

```diff
@@ -92,6 +92,17 @@ void OptimizeLatticeForCellType::exec() {
     }
   runWS.emplace_back(ws);
 
+  int maxOrder = ws->mutableSample().getOrientedLattice().getMaxOrder();
+  DblMatrix modHKL = ws->mutableSample().getOrientedLattice().getModHKL();
+
+  if (maxOrder > 0) {
+    for (int i = 0; i < ws->getNumberPeaks(); i++) {
+      IPeak &peak = ws->getPeak(i);
+      V3D HKL = peak.getIntHKL() + modHKL * peak.getIntMNP();
```

Comment:
So this would not work if the HKL had been set by hand - but I believe the HKL are all that is required for the lattice to be optimized (i.e. you don't need to know the parent peak)  - should the re-indexing be optional? How are the MNP of found peaks set in your workflow?

Reply:
Using `PredictSatellitePeaks` or `IndexPeaks` gives the correct hklmnp vector in our workflows. This line only happens if the max order is greater than zero, so if you do set it by hand with max order equal to zero, I believe it will still work.

Reply:
I see, I hadn't set `SaveModulationInfo=True` in `IndexPeaks sorry!

Reply:
It seems that `IndexPEaks` does not set H,K,L to the modulated vector if RoundHKLs=True even if they were provided. Would this be necessary if it did? And should the behaviour in `IndexPEaks be fixed (happy to do in a separate PR)

Reply:
So you mean it rounds it to integer? If so, that should also be fixed because that doesn't make sense. I interpret `RoundHKLs` to mean to round to `nice` values. That means integer for nuclear only but includes the fractional offset for modulated peaks.

Reply:
I have created a separate issue for  `IndexPeaks` here #35513 - essentially it rounds the modulated peaks to have integer HKL if `RoundHKLS=True` - whereas I would expect the behaviour to set HKL to `HKL = peak.getIntHKL() + modHKL * peak.getIntMNP()` (as you have elsewhere in this PR).